Example/Tests/NetworkTests/UploaderTests/UploaderTests.swift

```diff
@@ -53,6 +53,118 @@ class UploaderTests: NetworkBaseTest {
         XCTAssertNotNil(result, "result should not be nil")
         XCTAssertNil(error, "error should be nil")
     }
+    
+    func testUploadPNGImageData() {
+
+        XCTAssertNotNil(cloudinary!.config.apiSecret, "Must set api secret for this test")
+
+        let expectation = self.expectation(description: "Upload should succeed")
+        let data = TestResourceType.logo.data
+
+        var result: CLDUploadResult?
+        var error: NSError?
+
+        let params = CLDUploadRequestParams()
+        params.setResourceType(.image)
+        let chain = CLDImagePreprocessChain().setEncoder(CLDPreprocessHelpers.customImageEncoder(format: EncodingFormat.PNG, quality: 100))
+        
+        cloudinary!.createUploader().signedUpload(data: data, params: params, preprocessChain: chain).response({ (resultRes, errorRes) in
+            result = resultRes
+            error = errorRes
+
+            expectation.fulfill()
+        })
+
+        waitForExpectations(timeout: timeout, handler: nil)
+
+        XCTAssertNotNil(result, "result should not be nil")
```

Comment:
@AlexPinhasov , Let's add a few more checks, similar to the last test. here we can check that the file size is as expected.